### ISSUE TITLE
(typescript): Request for adding different types(interfaces) for the query data from useQuery and subscribeToMore data

### ISSUE THREAD

**Issue by arkajyotiMukherjee** [apollographql/apollo-feature-requests#256]
```
const { subscribeToMore, data, loading, error } = useQuery<MyQueryInterface>(
   MY_QUERY_DOCUMENT
  );
```
The `MyQueryInterface` defines the variable `data`'s type.
Now I have a use-case where the data from the Query is an array and the subscribeToMore subscription returns me any new array element that are created.
For example, the interface for query might be 
```
interface MyQueryInterface {
  someArray: number[];
}
```
but my subscription data returns
```
interface MyQueryInterface {
  singleArrayElement: number;
}
```
Now in the `subscribeToMore` function I have to combine these two and it works fine if I do not type my query but if I do type them the subscription data doesn't have a `someArray` and it throws a type error.

If I could provide types for my subscription in `subscribeToMore` like I can give types to my query data and query variables...that would be nice.
Like:
```
const { subscribeToMore, data, loading, error } = useQuery<
    MyQueryInterface, MySubsriptionInterface, QueryVriableInterface
  >(
   MY_QUERY_DOCUMENT
  );
```

Do let me know if this feature is a duplicate or is not valid...I'll provide some more context.
Thanks in advance.

**Comment by mahieyin-rahmun** [comment]
I agree that this should be in the docs. I was struggling with this until I found this issue. Thanks for sharing the solution/workaround!

**Comment by jpvajda** [comment]
@arkajyotiMukherjee thanks for reporting this, I'll move this over to the Apollo Client project and determine if we can make a documentation improvement on this topic.
